src/dmd/dcast.d

```diff
@@ -1456,6 +1456,42 @@ MATCH implicitConvTo(Expression e, Type t)
     return v.result;
 }
 
+/**
+ * Same as implicitConvTo(); except follow C11 rules, which are quite a bit
+ * more permissive than D.
+ * C11 6.3 and 6.5.16.1
+ * Params:
+ *   e = Expression that is to be casted
+ *   t = Expected resulting type
+ * Returns:
+ *   The `MATCH` level between `e.type` and `t`.
+ */
+MATCH cimplicitConvTo(Expression e, Type t)
```

Comment:
Is it wise to mix C-analysis functions with D-analysis functions? This may impact packaging the compiler as a library

Reply:
I don't really know. A pattern may emerge after some more of this. In the meantime, I've segregated such into separate functions. A thought is to eventually create a importc.d file and move them there.